_data-prepper/pipelines/configuration/sources/kafka.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: default
+title: kafka
+parent: Sources
+grand_parent: Pipelines
+nav_order: 6
+---
+
+# kafka
+
+The `kafka` source allows Data Prepper to use Apache Kafka as a source. reads records from one of more Kafka [topics](https://kafka.apache.org/intro#intro_concepts_and_terms), which hold events that your Data Prepper pipeline can ingest. It uses Kafka's [Consumer API](https://kafka.apache.org/documentation/#consumerapi) to consume messages from the Kafka broker, which then creates Data Prepper events for further processing by the Data Prepper pipeline.
+
+## Usage
+
+The following example shows the `kafka` source in a Data Prepper pipeline:
+
+```json
+kafka-pipeline:
+  source:
+    kafka:
+      bootstrap_servers:
+        - 127.0.0.1:9093
+      topics:
+        - name: Topic1
+          group_id: groupID1
+        - name: Topic2
+          group_id: groupID1
+```
+
+## Configuration
+
+Use the following configuration options with the `kafka` source.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`bootstrap_servers` | Yes when not using Amazon MSK as a cluster | IP address | The host or port for the initial connection to the Kafka cluster. You can configure multiple Kafka brokers by using the IP address or port number for each broker. When using [Amazon MSK](https://aws.amazon.com/msk/) as your Kafka cluster, the bootstrap server information is obtained from MSK using the MSK ARN provided in the configuration.
+`topics` | Yes | JSON array | The topics inside of Kafka where the Data Prepper `kafka` source reads messages. You can configure up to 10 topics. For more information about options you need to configure inside `topics`, see [Topics](#topics).
```

Comment:
Confirm that my changes retain the intended meaning. If not, let's discuss.

Reply:
They do